django/template/base.py

```diff
@@ -227,6 +227,7 @@ def compile_nodelist(self):
             tokens, self.engine.template_libraries, self.engine.template_builtins,
             self.origin,
         )
+        parser.template_name = self.origin.template_name
```

Comment:
Initualy this code was wrote for Django 1.4, that don't has a 'origin' attribute into parser. 10.0 can use parser.origin.template_name of course.